Assets/Scripts/Dialogue/InkManager.cs

```diff
@@ -173,17 +169,56 @@ private string GetSpeakerTagValue(string pTag)
             return "Given tag isn't a speaker tag";
         } 
     }
+    
+    private string GetEmotionTagValue(string pTag)
+    {
+        if (pTag.Contains("emotion"))
+        {
+            string[] speakerTagContent = pTag.Split(':');
 
-    private void PortraitSetup()
+            if (speakerTagContent.Length != 2) Debug.LogError("Emotion tag can't be read", this);
+
+            return speakerTagContent[1];
+        }
+        else
+        {
+            Debug.LogError("Given tag isn't an emotion tag", this);
+            return "Given tag isn't an emotion tag";
+        } 
+    }
+
+    /// <summary>
+    /// A is the player character, B is the character they're talking to
+    /// </summary>
+    /// <param name="pSpeakerID"></param>
+    private void SetName(string pSpeakerID)
     {
-        playerImage.sprite = dialogue.CharacterASprite;
-        npcImage.sprite = dialogue.CharacterBSprite;
+        if (pSpeakerID == "A")
```

Comment:
Can you make "A" a const variable? Then it's easier to change the value if necessary.